Frontend/src/app/home/filter-dialog/filter-dialog.component.html

```diff
@@ -0,0 +1,35 @@
+<div [formGroup]="formGroup" mat-dialog-content class="dialog-container">
+  <h1>Filter restaurants</h1>
+  <div>
+    <h2>Price Category:</h2>
+    <mat-form-field class="form-field" appearance="outline">
+      <mat-select class="select" formControlName="priceCategoryControl" multiple
+                  [value]="formGroup.get('priceCategoryControl')">
+        @for (priceCategory of priceCategories; track priceCategory) {
+          <mat-option class="option" [value]="priceCategory">
+            @for (_ of [].constructor(priceCategory); track $index) {
+              $
+            }
+          </mat-option>
+        }
+      </mat-select>
+    </mat-form-field>
+  </div>
+  <div>
+    <h2>Food Category:</h2>
+    <mat-form-field class="form-field" appearance="outline">
+      <mat-select class="select" formControlName="foodCategoryControl" multiple
+                  [value]="formGroup.get('foodCategoryControl')">
+        @for (foodCategory of foodCategories; track foodCategory) {
+          <mat-option class="option" [value]="foodCategory">{{ foodCategory }}</mat-option>
+        }
+      </mat-select>
+    </mat-form-field>
+  </div>
+
+  <mat-dialog-actions class="dialog-actions">
+    <button mat-raised-button (click)="onFilterClick()">Filter</button>
```

Comment:
This should by type="submit" and instead of calling `onFilterClick()` here call it in the `(submit)` or `(ngSubmit)` event of the form (the div should a form control)